Remotion/Data/DomainObjects/Mapping/ClassDefinition.cs

```diff
@@ -120,66 +90,52 @@ public bool IsSameOrBaseClassOf (ClassDefinition classDefinition)
       return false;
     }
 
+    public override TypeDefinition[] GetTypeHierarchy ()
+    {
+      var allDerivedClasses = new List<ClassDefinition>();
+      allDerivedClasses.Add(this);
+      FillAllDerivedClasses(allDerivedClasses);
+
+      var allDerivedTypes = new TypeDefinition[allDerivedClasses.Count];
+      for (var i = 0; i < allDerivedTypes.Length; i++)
+        allDerivedTypes[i] = allDerivedClasses[i];
+
+      return allDerivedTypes;
+    }
+
     public ClassDefinition[] GetAllDerivedClasses ()
     {
       var allDerivedClasses = new List<ClassDefinition>();
       FillAllDerivedClasses(allDerivedClasses);
       return allDerivedClasses.ToArray();
     }
 
-    public ClassDefinition GetInheritanceRootClass ()
+    public override TypeDefinition GetInheritanceRoot ()
     {
       if (BaseClass != null)
-        return BaseClass.GetInheritanceRootClass();
+        return BaseClass.GetInheritanceRoot();
 
       return this;
     }
 
-    [Obsolete("Contains (...) method was ambiguous between GetPropertyDefinitions() and MyPropertyDefinitions. Use GetPropertyDefinitions().Contains (...) instead. (Version: 3.0.0)", true)]
+    [Obsolete(
+        "Contains (...) method was ambiguous between GetPropertyDefinitions() and MyPropertyDefinitions. Use GetPropertyDefinitions().Contains (...) instead. (Version: 3.0.0)",
+        true)]
     public bool Contains (PropertyDefinition propertyDefinition)
```

Comment:
should get moved to base type. the obsolete-method's just there to inform the developer what happend to the feature, so when we now swtich from class to type, the developer will be looking for thsi on TypeDef.